### PR TITLE
implement overwrite_input in cupy.{percentile,quantile}

### PR DESCRIPTION
**Pull request by saswatpp**
Addresses #6256. 
Signature of `cupy.{percentile,quantile}` was changed to match that of NumPy API. 
Tests also added, for both. 

@kmaehashi

### CODE REVIEW

File: tests/cupy_tests/statistics_tests/test_order.py [hunk 1]
```diff
@@ -143,6 +143,14 @@ def test_percentile_out(self, xp, dtype, method):
         return xp.percentile(
             a, q, axis=-1, method=method, out=out)
 
+    @testing.for_float_dtypes(no_float16=True)
+    @testing.numpy_cupy_allclose(rtol=1e-6)
+    def test_percentile_overwrite(self, xp, dtype, method):
+        a = testing.shaped_random((10, 2, 3, 2), xp, dtype)
+        q = testing.shaped_random((5,), xp, dtype=dtype, scale=100)
+        return xp.percentile(
+            a, q, axis=-1, method=method, overwrite_input=True)
```
Comment (emcastillo) [inline tests/cupy_tests/statistics_tests/test_order.py:152]: I think you need to assert here that the contents of a have changed wrt the original ones. :)

Comment (saswatpp) [reply tests/cupy_tests/statistics_tests/test_order.py:152]: Thank you for the Review :) Is the following okay ?

```python3
a = testing.shaped_random((10, 2, 3, 2), xp, dtype)
ap = a.copy()
q = testing.shaped_random((5,), xp, dtype=dtype, scale=100)

res = xp.percentile(
        ap, q, axis=-1, method=method, overwrite_input=True)

assert not xp.all(ap == a)
return res
```

Comment (emcastillo) [reply tests/cupy_tests/statistics_tests/test_order.py:152]: yeah, this looks good!

Comment (saswatpp) [reply tests/cupy_tests/statistics_tests/test_order.py:152]: Ok

File: cupy/_statistics/order.py [hunk 2]
```diff
@@ -283,7 +290,10 @@ def _quantile_is_valid(q):
     return True
 
 
-def percentile(a, q, axis=None, out=None, *, method='linear', keepdims=False,
+def percentile(a, q, axis=None, out=None, *,
+               overwrite_input=False,
+               method='linear',
+               keepdims=False,
```
Comment (kmaehashi) [inline cupy/_statistics/order.py:296]: Align with NumPy's signature.

```suggestion
def percentile(a, q, axis=None, out=None,
               overwrite_input=False,
               method='linear',
               keepdims=False,
               *,
```

File: cupy/_statistics/order.py [hunk 3]
```diff
@@ -316,10 +330,16 @@ def percentile(a, q, axis=None, out=None, *, method='linear', keepdims=False,
...
+        overwrite_input=overwrite_input,
+        method=method,
+        keepdims=keepdims)
 
 
-def quantile(a, q, axis=None, out=None, *, method='linear', keepdims=False,
+def quantile(a, q, axis=None, out=None, *,
+             overwrite_input=False,
+             method='linear',
+             keepdims=False,
```
Comment (kmaehashi) [inline cupy/_statistics/order.py:342]:
```suggestion
def quantile(a, q, axis=None, out=None,
             overwrite_input=False,
             method='linear',
             keepdims=False,
             *,
```

**Review by kmaehashi — CHANGES_REQUESTED** [review]

**Review by emcastillo — APPROVED** [review]

**Review by kmaehashi — APPROVED** [review]

**Comment by kmaehashi** [comment]
@emcastillo I guess you forgot enabling auto-merge 😉

**Pull request merged**